tests/e2e/certrotation/certrotation_test.go

```diff
@@ -0,0 +1,112 @@
+
+var _ = Describe("Verify Custom CA Rotation", Ordered, func() {
+	Context("Custom CA is rotated:", func() {
+		It("Starts up with no issues", func() {
+			var err error
+			serverNodeNames, agentNodeNames, err = e2e.CreateCluster(*nodeOS, *serverCount, *agentCount)
+			Expect(err).NotTo(HaveOccurred(), e2e.GetVagrantLog(err))
+			fmt.Println("CLUSTER CONFIG")
+			fmt.Println("OS:", *nodeOS)
+			fmt.Println("Server Nodes:", serverNodeNames)
+			fmt.Println("Agent Nodes:", agentNodeNames)
+			kubeConfigFile, err = e2e.GenKubeConfigFile(serverNodeNames[0])
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("Verifies Certificate Rotation", func() {
+			const grepCert = "sudo ls -lt /var/lib/rancher/k3s/server/ | grep tls"
+			var expectResult = []string{"client-ca.crt", "client-ca.key", "client-ca.nochain.crt", "client-ca.pem",
+				"dynamic-cert.json", "peer-ca.crt", "peer-ca.key", "peer-ca.pem", "server-ca.crt", "server-ca.key", "server-ca.pem",
+				"intermediate-ca.crt", "intermediate-ca.key",
+				"intermediate-ca.pem", "request-header-ca.crt", "request-header-ca.key",
+				"request-header-ca.pem", "root-ca.crt", "root-ca.key", "root-ca.pem", "server-ca.crt", "server-ca.key",
+				"server-ca.nochain.crt", "server-ca.pem", "service.current.key", "service.key", "apiserver-loopback-client__.crt",
+				"apiserver-loopback-client__.key", "",
+			}
+
+			var finalResult string
+			var finalErr error
+			errStop := e2e.StopCluster(serverNodeNames)
+			Expect(errStop).NotTo(HaveOccurred(), "Server not stop correctly")
+			errRotate := e2e.RotateCertificate(serverNodeNames)
+			Expect(errRotate).NotTo(HaveOccurred(), "Certificate not rotate correctly")
+			errStart := e2e.StartCluster(serverNodeNames)
+			Expect(errStart).NotTo(HaveOccurred(), "Server not start correctly")
+
+			for _, nodeName := range serverNodeNames {
+				grCert, errGrep := e2e.RunCmdOnNode(grepCert, nodeName)
+				Expect(errGrep).NotTo(HaveOccurred(), "Certificate not created correctly")
+				re := regexp.MustCompile("tls-[0-9]+")
+				tls := re.FindAllString(grCert, -1)[0]
+				final := fmt.Sprintf("sudo diff -sr /var/lib/rancher/k3s/server/tls/ /var/lib/rancher/k3s/server/%s/ | grep -i identical | cut -f4 -d ' ' | xargs basename -a \n", tls)
+				finalResult, finalErr = e2e.RunCmdOnNode(final, nodeName)
+				Expect(finalErr).NotTo(HaveOccurred(), "Final Certification does not created correctly")
+			}
+			if len(agentNodeNames) > 0 {
+				errRestartAgent := e2e.RestartCluster(agentNodeNames)
+				Expect(errRestartAgent).NotTo(HaveOccurred(), "Restart Agent not happened correctly")
+			}
+			Eventually(func(g Gomega) {
+				finalCert := strings.Replace(finalResult, "\n", ",", -1)
+				finalCertArray := strings.Split(finalCert, ",")
+				fmt.Println("FINAL ESTEBAN ARRAY", finalCertArray)
+				fmt.Println("FINAL ESTEBAN expectResult", expectResult)
+				Expect((finalCertArray)).Should((Equal(expectResult)), "Final certification does not match the expected results")
```

Comment:
here is correct Should(Equal) ? or should not ?

Reply:
Yes!, it is correct to compare the Certificate generate against the expect result.